src/controllers/baseHandler.ts

```diff
@@ -42,9 +46,27 @@ export async function baseHandler(
         roleToBeTaggedObj: data[0],
         displayMessageObj: data[1] ?? {},
       };
-
       return await mentionEachUser(transformedArgument, env);
     }
+
+    case getCommandName(LISTENING): {
+      const data = message.data?.options;
+      const setter = data ? data[0].value : false;
+      const nickname = removeListening(message.member.nick || "");
+      if (setter) {
+        if (!message.member.nick?.includes(LISTENING_COPY)) {
+          await updateNickName(
+            `${message.member.user.id}`,
+            "🎧 " + message.member.nick + LISTENING_COPY,
+            env
+          );
+        }
+      } else {
+        await updateNickName(`${message.member.user.id}`, nickname, env);
+      }
+
+      return discordEphemeralResponse(DESC_MESSAGE);
```

Comment:
Please provide appropriate message such as 
"Your name is now changed to reflect your listening status"
in case of removal al well lets provide an appropriate message.